src/physical/datatypes/storage_value.rs

```diff
@@ -24,30 +26,39 @@ pub enum StorageValueT {
 
 impl StorageValueT {
     /// Returns either [`Option<u32>`], answering whether the [`StorageValueT`] is of this datatype
-    pub fn as_u32(&self) -> Option<u32> {
+    pub fn get_u32(&self) -> Option<u32> {
```

Comment:
The more I think about it, I think it would be even better to get rid of these entirely and replace them with `TryFrom` implementations on the underlying types.

Reply:
The reason was that I am confused when `as_` prefixed functions do not return a borrow (which is also stated in the API naming guidelines). Thinking about this some more though, the real confusion comes from the fact, that these do _two_ operations in one: Copy + TryInto. Implementing `TryFrom` is probably forbidden due to rusts orphaning rules (we do not own the trait nor the `u32` type).

Reply:
How about
```suggestion
    pub fn try_into_u32(self) -> Option<u32> {
```

Reply:
> Implementing TryFrom is probably forbidden due to rusts orphaning rules (we do not own the trait nor the u32 type).
Actually it is not. So I will go ahead and do this instead.

Reply:
I vaguely remember this being forbidden in old Rust versions, but it appears that since [Rust 1.41](https://blog.rust-lang.org/2020/01/30/Rust-1.41.0.html#relaxed-restrictions-when-implementing-traits), it's sufficient that the generic parameter is crate-local.